tests/pkg_libbase58/main.c

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2019 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup    tests
+ * @{
+ *
+ * @file
+ * @brief      Tests for pkg libbase58
+ *
+ * @author     Kaspar Schleiser <kaspar@schleiser.de>
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdint.h>
+
+#include "libbase58.h"
+#include "embUnit.h"
+
+static void setUp(void)
+{
+}
+
+static void test_libbase58_01(void)
+{
+    const char source[] = "base 58 test string";
+    const char encoded[] = "K1cpEhnSEL8mbBfpp7f6z3dSZ4";
+    char target[64] = {0};
+    size_t target_len = sizeof(target);
+
+    b58enc(target, &target_len, source, sizeof(source)-1);
```

Comment:
Questions for testing:

- What about decoding tests (also a doc question: is there a decoding function)?
- What happens if I throw in a NULL pointer?
- What happens if the string is empty?
- Does it still work if I provide a source length that `source` actually is?

Reply:
Also, where did your test data came from? How can I verify it?

Reply:
I've added a decoding step to the tests. For everything else, I'd suggest sticking to the libbase58 documentation.

> Also, where did your test data came from? How can I verify it?

I've added a "puts" at strategic places and then copy&pasted the string from the (at that point failing) test output...